test/suite/test_prefetch02.py

```diff
@@ -0,0 +1,143 @@
+
+    def get_stat(self, stat, session_name):
+        stat_cursor = session_name.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def get_prefetch_activity_stats(self, session_name):
+        pages_queued = self.get_stat(wiredtiger.stat.conn.block_prefetch_pages_queued, session_name)
+        prefetch_attempts = self.get_stat(wiredtiger.stat.conn.block_prefetch_attempts, session_name)
+        prefetch_pages_read = self.get_stat(wiredtiger.stat.conn.block_prefetch_pages_read, session_name)
+        return pages_queued, prefetch_attempts, prefetch_pages_read
+
+    def get_prefetch_skipped_stat(self, session_name):
+        prefetch_skips = self.get_stat(wiredtiger.stat.conn.block_prefetch_skipped, session_name)
+        return prefetch_skips
+
+    # Checks for pre-fetching activity by asserting that relevant statistics have increased.
+    def check_prefetching_activity(self, session_name, pages_queued, prefetch_attempts, prefetch_pages_read):
+        new_pages_queued, new_prefetch_attempts, new_prefetch_pages_read = self.get_prefetch_activity_stats(session_name)
+        self.assertGreaterEqual(new_pages_queued, pages_queued)
+        self.assertGreater(new_prefetch_attempts, prefetch_attempts)
+        self.assertGreaterEqual(new_prefetch_pages_read, prefetch_pages_read)
+
+    # Checks that the values of statistics related to pre-fetching activity are equal to zero,
+    # and that pages are being skipped when deciding whether to pre-fetch the page or not.
+    def check_no_prefetching_activity(self, session_name, prefetch_skips):
+        new_prefetch_skips = self.get_prefetch_skipped_stat(session_name)
+        pages_queued, prefetch_attempts, prefetch_pages_read = self.get_prefetch_activity_stats(session_name)
+        self.assertGreaterEqual(new_prefetch_skips, prefetch_skips)
+        self.assertEqual(pages_queued, 0)
+        self.assertEqual(prefetch_attempts, 0)
+        self.assertEqual(prefetch_pages_read, 0)
+
+    def test_prefetch_scenarios(self):
+        os.mkdir(self.new_dir)
+        helper.copy_wiredtiger_home(self, '.', self.new_dir)
+
+        new_conn = self.wiredtiger_open(self.new_dir, self.conn_cfg)
+        s = new_conn.open_session(self.session_cfg)
+        s.create(self.uri, 'key_format=i,value_format=i')
+        c1 = s.open_cursor(self.uri)
+        s.begin_transaction()
+        for i in range(100000):
+            c1[i] = i
+        c1.close()
+        s.commit_transaction()
+        s.checkpoint()
+
+        if self.scenario_type == 'traversal':
+            c2 = s.open_cursor(self.uri)
+
+            # Traverse through half the key space and collect pre-fetching statistics. Then, traverse
+            # through the rest of the keys and check that the relevant pre-fetching statistics have
+            # increased by the end. If pre-fetching is not available, check that we are skipping pages.
+            for i in range(50000):
```

Comment:
Could you please get rid of some of the magic numbers? the 100,000 and the 50,000 would be clearer if it was called `nrows` or something.